### PR TITLE
Custom folders

### PR DESCRIPTION
**Pull request by czclaxton**
# Description
created crud for custom folders

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change status

- [x] Complete, tested, ready to review and merge
- [ ] Work-in-Progress, PR is for discussion/feedback

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] There are no merge conflicts
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by luiscmartinez — APPROVED** [review]
A 1 ++++ PR 👽👽👽👽☠🧐

File: components/auth/authRoutes.js [hunk 1]
```diff
@@ -25,24 +27,37 @@ router.get(
...
+/*  ================== GOODREADS ================== */
 router.get('/goodreads', passport.authenticate('goodreads'))
 
+// allow developer to generate a token 4 REST client usage
+if (process.env.NODE_ENV === 'development') {
+  router.get('/token', async (req, res) => {
+    /*  ================== Make sure to run the seeds! ================== */
+    const user = await db('users')
+      .where({ id: 1 })
+      .first()
```
Comment (cesarnml) [inline components/auth/authRoutes.js]:
```suggestion
    const user = await db('users')
      .where({ id: process.env.DEV_USER_ID })
      .first()
```

@luiscmartinez 
Don't we want this to be dynamic? So we can test via postman using our own dev user.  `user_id = 1` will be a fake user.

File: components/auth/authRoutes.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ const xpath = require('xpath')
 const dom = require('xmldom').DOMParser
 const axios = require('axios')
 const db = require('../../dbConfig')
+const generateToken = require('../../utils').generateToken
```
Comment (cesarnml) [inline components/auth/authRoutes.js:8]:
```suggestion
const generateToken = require('../../utils').generateToken
require('dotenv').config() // Need access to process.env.DEV_USER_ID
```
see `line 47-49`

File: components/folders/folderRoutes.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
 const router = require('express').Router()
 const controllers = require('./folderControllers')
+const utils = require('../../utils')
 
-router.post('/', controllers.createFolder)
+router.post('/', utils.isRequestAuthenticated, controllers.createFolder)
```
Comment (cesarnml) [inline components/folders/folderRoutes.js:5]: `router.post('/', utils.isRequestAuthenticated, controllers.createFolder)`
@czclaxton @luiscmartinez 

Okay for now, but need to move away from authenticating on each route. This should be a middleware that is called once before entering any protected route. We e only want to protect routes that write/update database or protect sensitive information.

**Review by cesarnml — APPROVED** [review]
@luiscmartinez 

LGTM: I made a few comments, but this is ready to merge after `token` is updated to be associated with a dynamic `user_id` via `process.env.DEV_USER_ID`

Merge when ready.

ENV documentation should be updated ...

**Pull request merged**